commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPIHandler.java

```diff
@@ -358,30 +358,35 @@ Command<CommandSourceStack> generateCommand(Argument<?>[] args,
 	 * @return an Object[] which can be used in (sender, args) ->
 	 * @throws CommandSyntaxException
 	 */
-	Object[] argsToObjectArr(CommandContext<CommandSourceStack> cmdCtx, Argument<?>[] args)
+	Map<Integer, Object> argsToObjectArr(CommandContext<CommandSourceStack> cmdCtx, Argument<?>[] args)
 			throws CommandSyntaxException {
 		// Array for arguments for executor
 		List<Object> argList = new ArrayList<>();
 
+		// LinkedHashMap for arguments for executor
+		Map<String, Object> argsMap = new LinkedHashMap<>();
+
 		// Populate array
 		for (Argument<?> argument : args) {
 			if (argument.isListed()) {
 				argList.add(parseArgument(cmdCtx, argument.getNodeName(), argument, argList.toArray()));
+				argsMap.put(argument.getNodeName(), parseArgument(cmdCtx, argument.getNodeName(), argument, argList.toArray()));
 			}
 		}
 
-		return argList.toArray();
+		Map<Integer, Object> arguments = new HashMap<>();
```

Comment:
This looks sus. What's going on here? By the looks of things, it looks like a really weird way to implement "returning two objects at the same time". Surely here we're at a point we can just return a new `CommandArguments` object?